### DISCUSSION TITLE
How to receive binary object in nodejs custom plugin lib

### DISCUSSION THREAD

**Discussion by elmalakomar** [mia-platform/community#163] [Q&A] [answered]
Hi, 
what is the best approach to retrieve a binary object in the body of a nodejs custom plugin lib endpoint while the are other fields in the body? What is the json schema type of the binary object? a string? 

example of a body
```
{
 fileContent: string
 otherStringField: string
}
```

We are trying this approach:
```
export const fastifyPluginCallback: FastifyPluginCallback = async service => {
  service.addRawCustomPlugin('POST', '/endpoint-file', fileHandler, fileSchema)

  service.addContentTypeParser('multipart/form-data', { parseAs: 'string' }, service.getDefaultJsonParser('ignore', 'ignore'))
}
```

Thank you!

**Comment by fredmaggiowski** [comment, marked as answer]
Hi there.

I'm not sure if I got your question correctly; you need to provide a binary content in your API however do you need the body to be binary-only or you need other data to be provided as well?

In the first case I'd suggest you to simply use the `multipart/form-data` Content Type and verify the proper configuration for the json schema provided by fastify; in the second case you have two options:

1. using `multipart/form-data` for transmitting both the binary content AND some data in `form-urlencoded` fashion, or
1. using `application/json` Content Type and have your binary content encoded in base64 as a value for some specific key


I'd base the choice on how complex/extensible you want the API interaction to be with your client (the json + base64 value seems pretty easy to implement in any client-side language and reasonably extensible)
